core/src/main/scala/kafka/log/Log.scala

```diff
@@ -1139,9 +1139,13 @@ object Log {
       throw exception(dir)
 
     val name: String =
-      if (dirName.endsWith(DeleteDirSuffix)) dirName.substring(0, dirName.indexOf('.'))
+      if (dirName.endsWith(DeleteDirSuffix)) dirName.substring(0, dirName.lastIndexOf('.'))
       else dirName
 
+    if (!name.contains('-')) {
```

Comment:
@ijuma, I had a look at the code in trunk and as you note, this additional check won't be necessary there. I will raise a separate PR for trunk which won't include this additional check.